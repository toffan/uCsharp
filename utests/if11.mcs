int main() {
    int c1 = 3;
    int c2 = 0;

    if (c1) {
        int a = 5;
    } else {
        if (c2) {
            int b = 2;
        } else {
            int a = 3;
        }
    }

    return 0;
}
