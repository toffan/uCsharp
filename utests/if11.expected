JUMP __main
_main:
    LOADL 3
    LOADL 0
    LOAD (1) 3[LB]
    SUBR I2B
    JUMPIF (0) ELSE_1
    LOADL 5
    POP (0) 1
    JUMP ENDIF_1
    ELSE_1:
    LOAD (1) 4[LB]
    SUBR I2B
    JUMPIF (0) ELSE_0
    LOADL 2
    POP (0) 1
    JUMP ENDIF_0
    ELSE_0:
    LOADL 3
    POP (0) 1
    ENDIF_0:
    POP (0) 0
    ENDIF_1:

    LOADL 0
    RETURN (1) 0
    RETURN (0) 0
__main:
    CALL (SB) _main
    HALT
